### PR TITLE
json/outbound: test request and response headers

### PR DESCRIPTION
**Pull request by abhinav**
CC @yarpc/golang

### CODE REVIEW

**Comment by breerly** [comment]
nice :shipit:

File: encoding/json/outbound_test.go [hunk 1]
```diff
@@ -83,6 +85,16 @@ func TestCall(t *testing.T) {
...
 		},
+		{
+			procedure:       "requestHeaders",
+			headers:         transport.Headers{"user-id": "42"},
+			returnHeaders:   transport.Headers{"success": "true"},
+			body:            map[string]interface{}{},
+			encodedRequest:  "{}",
+			encodedResponse: "{}",
+			want:            map[string]interface{}{},
+			wantHeaders:     transport.Headers{"success": "true"},
```
Comment (prashantv) [inline encoding/json/outbound_test.go]: are `wantHeaders` always the same as `returnHeaders`?

**Comment by prashantv** [comment]
minor comment: since you're already using `tt.headers` to match, I think using `tt.returnHeaders` for the response headers assert is fine.

lgtm otherwise.

File: encoding/json/outbound_test.go (hunk 1 above)
Comment (abhinav) [inline encoding/json/outbound_test.go]: oh true

**Pull request merged**